src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -955,7 +973,11 @@ private SafeProcessHandle GetProcessHandle(int access, bool throwIfExited)
                         }
                     }
                 }
-                return _processHandle;
+
+                // If we dispose of our contained handle we'll be in a bad state. NetFX dealt with this
+                // by doing a try..finally around every usage of GetProcessHandle and only disposed if
```

Comment:
Because NetFX never put the handles in a using block. The implementation here does. I thought about matching the desktop pattern (e.g. removing the usings), but this seemed like a safer/smarter way to go. The desktop way is asking for  mistakes.